pkg/yamltemplate/template.go

```diff
@@ -93,6 +93,14 @@ func (e *Template) build(val interface{}, parentNode yamlmeta.Node, parentTag te
 		SourceLine:  e.newSourceLine(node.GetPosition()),
 	})
 
+	if typedNode, ok := val.(*yamlmeta.MapItem); ok {
+		templateLines, err := e.buildString(typedNode.Key.(string), parentNode, parentTag, e.instructions.NewSetMapItemKey)
```

Comment:
i think we also need to add `if keyStr, ok := typedNode.Key.(string); ok {` otherwise non-string keys will panic.